src/commands/removeCodeBlocks.ts

```diff
@@ -0,0 +1,470 @@
+        codeBlockDirectory,
+        codeBlockInstance,
+        langDetails.extension
+      );
+      // Initialize a code block type with text and metadata about the code block
+      const codeBlockWithMetadata: CodeBlockWithMetadata = {
+        language: langDetails.canonicalValue,
+        writeFilepath: codeBlockWriteFilePath,
+        literalincludeFilepath: literalincludeFilePath,
+        codeBlockDirectory: codeBlockDirectory,
+        content: formattedCodeBlockText,
+        optionLines: optionLines,
+      };
+      codeBlocks.push(codeBlockWithMetadata);
+
+      /* If there is an indent width, as when the code block is nested in a 'step' directive
+         or a 'tab' directive, calculate an offset for the start location to account for the indent
+      */
+      let literalIncludeStartIndentWidth = 0;
+      if (node["indent"] !== undefined) {
+        const indentDetails = node["indent"];
+        if (indentDetails?.width !== undefined) {
+          literalIncludeStartIndentWidth = indentDetails?.width;
+        }
+      }
+
+      // Construct literalinclude
+      const literalInclude = makeLiteralInclude(
+        codeBlockWithMetadata,
+        literalIncludeStartIndentWidth
+      );
+
+      // Replace original code block directive with literalinclude
+      let start = node.position.start.offset;
+      const end = node.position.end.offset;
+
+      if (literalIncludeStartIndentWidth > 0) {
+        // Subtract three from the indent width for the `.. ` at the beginning of the code-block directive
+        start = start + literalIncludeStartIndentWidth - 3;
+      }
+      document.overwrite(start, end, literalInclude);
+    });
+  } catch (error) {
+    console.error(error);
+    return new MagicString(source); // Return unedited document
+  }
+  if (codeBlocks.length > 0) {
+    const pageWriteData: PageWriteData = {
+      inputFilepath: pageFilepath,
+      absFilepathToSource: absFilepathToSource,
+      codeBlockDirectory: codeBlockDirectory,
+      codeBlocks: codeBlocks,
+    };
+    await writeCodeBlocksToFile(pageWriteData);
+  }
+  return document;
+};
+
+/**
+ * Remove the code block directive and option lines, and get only the code text itself
+ * @param inputString - The entirety of the code block directive, including the `.. code-block::` directive start and code content
+ * @param indentWidth - The start column of the code block text, to properly adjust the offset to un-indent nested content
+ */
+const getFormattedCodeBlockTextFromDirective = (
+  inputString: string,
+  indentWidth: number
+): string => {
+  // Split the input string into lines
+  const lines = inputString.split("\n");
+
+  let index = 0;
+
+  // Skip the initial `.. code-block::` line
+  if (lines[index].trimStart().startsWith(".. code-block::")) {
+    index++;
+  }
+
+  // Skip any `:option:` lines
+  while (index < lines.length && lines[index].trimStart().startsWith(":")) {
```

Comment:
[q] Can this not be read from the node itself? It seems a bit sketchy to try to deduce this from the line itself